[BREAKING] Code refactoring

frugan-dev opened the pull request:
## DONE

- dropped support for Laravel <= 8
- dropped support for PHP <= 8.1 (Laravel ^9 || ^10 already supports PHP 8.2 https://endoflife.date/laravel)
- updated all composer packages
- added shared method `getResponse()`
- fixed `$response` returned by `Cache::remember()` https://stackoverflow.com/a/74783008/3929620
- used closure in `$response->throw()`
- revised esceptions classes, now extend `Illuminate\Http\Client\RequestException`
- used `Http::withToken()` instead of `Http::withHeaders()`
- used `Http::withQueryParameters()` instead of `http_build_query()`
- added `Http::withOptions()` and `Http::baseUrl()` 
- renamed parameter `$type` to `$name`, as it is more consistent with the field used by Strapi
- renamed `$queryData` to `$queryParams`, as it is more consistent with `Http::withQueryParameters()`
- removed almost all parameters (e.g. `$sortKey`, `$sortOrder`, `$populate`, etc.) in favor of `$queryParams`, which allows greater control over the API parameters used
- now `$queryParams` also allows you to use pagination by offset values (`pagination[start]` and `pagination[limit]`), which previously conflicted with pagination by page values (`pagination[page]` and `pagination[limit]`) set directly within the method `collection()`
- added optional `$cacheTime` parameter to override the global value defined in `STRAPI_CACHE_TIME`  per-call
- removed the `collectionCount()` method, as the `/api/$name/count` entrypoint does not seem to exist anymore in Strapi v4 (?), and also the `pagination[withCount]` parameter exists and it's active by default
- removed the `entriesByField()` method as it is superfluous, because it is possible to make the same call with the `collection()` method (see examples in `README`)
- moved option `STRAPI_FULL_URLS` globally in `.env` file
- removed `$pluck` parameter as it is superfluous to the primary functionality of this package: everyone is free to alter the response as they wish
- incremented `config('strapi.pagination.limit')` from 20 to 25 (default Strapi `pagination[pageSize]`)
- revised the entire `README` file

## TODO

- update references in Strapi documentation https://github.com/strapi/documentation/blob/main/docusaurus/docs/dev-docs/integrations/laravel.md
- might it be a good idea to add a **F.A.Q.** or **Known issues** section on the `README`? this could be included in this section, useful for those who have a local development setup (e.g. mkcert):

```
# .env.local
HTTP_CLIENT_VERIFY_SSL=false
```
```php
// config/http.php
return [
    'client' => [
        'verify_ssl' => (bool) env('HTTP_CLIENT_VERIFY_SSL', true),
    ],
];
```

```php
// app/Providers/AppServiceProvider.php
use Illuminate\Support\Facades\Http;

public function boot(): void
{
    Http::globalOptions([
	'verify' => config('http.client.verify_ssl'),
    ]);
}
```

frugan-dev committed:
Upgrade to PHP v8.2 https://endoflife.date/laravel

frugan-dev committed:
[BREAKING] Code refactoring

frugan-dev committed:
Typo

frugan-dev committed:
Updated README

frugan-dev committed:
Updated README

frugan-dev committed:
Removed collectionCount() method

frugan-dev committed:
Updated README

frugan-dev committed:
dropped support to Laravel <= 8
updated all composer packages
restored STRAPI_URL
revised esceptions classes

frugan-dev committed:
Fixed $response returned by Cache::remember() https://stackoverflow.com/a/74783008/3929620

frugan-dev committed:
Updated composer

frugan-dev committed:
Fixed response closure

frugan-dev committed:
fix(convertToFullUrls): use collection map

frugan-dev committed:
docs: add $fullUrls param

frugan-dev committed:
fix(cache)!: replaced cacheKey encrypt() with toBase64()

Stubbs reviewed (approved):
Looks good to me. The new queryParams means I don't have to re-implement my changes that introduced filters to the collection method.

frugan-dev commented:
Hi @dbfx, is there any problem accepting this pull request?

frugan-dev commented:
> @frugan-dev We have forked this package over at [BRFCS/laravel-strapi](https://github.com/BRFCS/laravel-strapi) because it looks like this repo is deprecated and we needed Laravel 11 support. Feel free to contribute if you still use Laravel & Strapi. We're not on packagist yet though.

Hi @Stubbs , If you ask for my contribution, you should have included my changes in this PR in your fork..
Anyway, why waste our efforts if this repo already has a certain following and is also linked in the [Strapi v4 docs](https://docs-v4.strapi.io/dev-docs/integrations/laravel#install-the-laravel-strapi-laravel-package)?
Let's ask to @dbfx if he can add us as contributors to this repo, in case he doesn't have time to actively follow it, or not?

frugan-dev committed:
feat!: support to laravel ^11

- upgrade require-dev libs

frugan-dev committed:
chore: config typo

frugan-dev committed:
perf: type declaration in entry() method

frugan-dev committed:
feat: add STRAPI_DEBUG in config

frugan-dev committed:
docs: update reqs

frugan-dev committed:
docs: typo

frugan-dev committed:
feat: add $response->notFound() to avoid throwing exceptions

frugan-dev commented:
The question should be asked to @dbfx

frugan-dev committed:
feat: add support to Laravel 12

- add CACHE_DISABLED and CACHE_FOREVER constants
- refactor getResponse() method

frugan-dev committed:
docs: update reqs

dbfx commented:
Hi guys, apologies for the delay only saw this now. Have invited you!

frugan-dev committed:
fix: fix HTTP response

frugan-dev committed:
refactor!: add STRAPI_CACHE_TYPE

Stubbs commented:
> Hi guys, apologies for the delay only saw this now. Have invited you!

and I missed the invite, which has now expired 🤦
